Sources/CreateAPI/Generator/Generator+Schemas.swift

```diff
@@ -762,3 +743,19 @@ extension Generator {
         return spec.components.schemas[key]
     }
 }
+
+private extension OpenAPI.Discriminator {
+    var schemaMapping: [String: JSONReference<JSONSchema>]? {
+        get throws {
+            try mapping?.mapValues { value in
+                if value.hasPrefix("#") {
+                    return .internal(.path(.init(rawValue: value)))
+                } else if let url = URL(string: value) {
+                    return .external(url)
+                } else {
+                    throw GeneratorError("Expected mapping value '\(value)' to be a valid reference")
+                }
```

Comment:
This roughly matches the implementation in OpenAPIKit but I didn't pay too much attention to external references since we don't actually support them currently.